typedoc-redirect-legacy-docs-plugin.mjs

```diff
@@ -0,0 +1,78 @@
+// TypeDoc plugin to redirect legacy docs to the new ones
+// Ref: https://github.com/Gerrit0/typedoc-plugin-redirect/blob/main/src/plugin.ts (Copyright 2024 Gerrit Birkeland, MIT License)
+
+import { PageEvent } from 'typedoc';
+
+/**
+ * Replace the input path with the new path.
+ * @param {string} path The legacy path
+ * @returns The new path
+ */
+function replacePath(path) {
+  // Item pages
+  // Before: /packages/three-vrm-animation/docs/classes/VRMAnimation.html
+  // After: /docs/classes/three-vrm-materials-mtoon.MToonMaterial.html
+  const matchItems = path.match(/packages\/(.+)\/docs\/(classes|interfaces|functions|types|variables)\/(.+)/);
+  if (matchItems != null) {
+    const [_, packageName, kind, item] = matchItems;
+    return `../../../docs/${kind}/${packageName}.${item}`;
```

Comment:
The relative path `../../../` does not work for nested docs like `/packages/three-vrm/docs/variables/VRMExpressionMaterialColorType-1.html`. Maybe using `/docs/` is better?

Reply:
Oops that's true. will change it to simply use `/docs/...`.

Reply:
no that wouldn't work because the document root will be `https://pixiv.github.io/three-vrm/docs/` on GitHub Pages. I would try setting the document root using environment variable.

Reply:
make it use an environment variable `DOCS_ROOT`.

https://github.com/pixiv/three-vrm/pull/1588/commits/8f9f9ae309409bc2ce1badeef1df90cf9b0dc606